src/main/java/com/systemmeltdown/robotlog/outputs/ZipFileUtils.java

```diff
@@ -0,0 +1,163 @@
+    File zipFile = getFile(dir, fileName);
+
+    System.out.println("ZipFileOutput: Creating zip file '" + zipFile + "'");
+    FileOutputStream fileOut = new FileOutputStream(zipFile);
+    BufferedOutputStream bufferedOut = new BufferedOutputStream(fileOut);
+    return new ZipOutputStream(bufferedOut);
+  }
+
+  public static File getFile(File dir, String fileName) {
+    return getFile(dir, fileName, 0);
+  }
+
+  public static File getFile(File dir, String fileName, int count) {
+    String fullFileName = fileName + (count > 0 ? "." + count : "") + ZIP_EXTENSION;
+
+    File file = new File(dir, fullFileName);
+
+    if (file.exists()) {
+      return getFile(dir, fileName, count + 1);
+    }
+    return file;
+  }
+
+  public static File getOrCreateDir(String path) {
+    File dir = new File(path);
+
+    if (!dir.exists()) {
+      String parentPath = dir.getParent();
+
+      if (parentPath == null) {
+        System.err.println("ZipFileOutput: Cannot create dir (" + path + ") for unknown parent dir");
+        return null;
+      }
+
+      File parentDir = getOrCreateDir(parentPath);
+      if (parentDir == null) {
+        // No need to log to stderr here, the parent recursive call should have already logged.
+        return null;
+      }
+
+      System.out.println("ZipFileOutput: Creating directory '" + path + "'");
+      if (!dir.mkdir()) {
+        System.err.println("ZipFileOutput: Failed to create parent directory: " + parentPath);
+        return null;
+      }
+    }
+
+    if (!dir.isDirectory()) {
+      System.err.println("ZipFileOutput: Path is not directory: " + path);
+      return null;
+    }
+    if (!dir.canWrite() || !dir.canExecute()) {
+      System.err.println("ZipFileOutput: Insufficient privileges for directory: " + path);
+      return null;
+    }
+
+    // Everything checks out.
+    return dir;
+  }
+
+  public static String printHeader(Map<String, Object> header) {
+    StringBuffer sb = new StringBuffer();
+    boolean needsComma = false;
+
+    sb.append("\"header\": { ");
+
+    for (Map.Entry<String, Object> entry : header.entrySet()) {
+      String keyStr = "\"" + entry.getKey() + "\"";
+      String valueStr = "\"" + printValue(entry.getValue(), entry.getValue().getClass()) + "\"";
+
+      if (needsComma) {
+        sb.append(", ");
+      }
+      sb.append(keyStr + ": " + valueStr);
+      needsComma = true;
+    }
+
+    sb.append(" }");
+    return sb.toString();
+  }
+
+  public static String printTopic(String topicName, Class<?> valueType) {
+    String topicStr = "\"" + topicName + "\"";
+    String valueTypeStr = "\"" + printValueType(valueType) + "\"";
+
+    return "{ \"name\":" + topicStr + ", \"type\":" + valueTypeStr + " }";
+  }
+
+  public static String printEntry(String topicName, Object value, Class<?> valueType, double relativeNanos, double timeRoundingFactor) {
+    String topicStr = "\"" + topicName + "\"";
+    String valueStr = printValue(value, valueType);
```

Comment:
printValue doesn't quote strings, so this will cause problems with string values

Reply:
This should be fixed by moving the quotes to `printValue`.